lib/eslint.js

```diff
@@ -807,20 +811,18 @@ module.exports = (function() {
              */
             controller.traverse(ast, {
                 enter: function(node, parent) {
-
                     var comments = api.getComments(node);
 
                     emitCommentsEnter(comments.leading);
                     node.parent = parent;
-                    api.emit(node.type, node);
+                    codePathAnalyzer.enterNode(node);
```

Comment:
Actually I think option 3 will give us a very nice doorway for selectors too. We would just need to add SelectorAnalyzer (or whatever we decide to call that) at the same level as `CommentTraverser` and `CodePathAnalyzer`.